odl/tomo/backends/astra_cpu.py

```diff
@@ -0,0 +1,192 @@
+        raise ValueError('dimensions {} of volume data and {} of geometry '
+                         'do not match.'
+                         ''.format(vol_data.space.grid.ndim, geometry.ndim))
+
+    ndim = vol_data.space.grid.ndim
+
+    # Create astra geometries
+    vol_geom = astra_volume_geometry(vol_data.space)
+    # proj_geom = astra_projection_geometry(geometry, vol_data.space)
+    proj_geom = astra_projection_geometry(geometry)
+
+    if out is None:
+        out = proj_space.element()
+
+    # Create ASTRA data structures
+    vol_id = astra_data(vol_geom, datatype='volume', data=vol_data)
+    sino_id = astra_data(proj_geom, datatype='projection', data=out,
+                         ndim=proj_space.grid.ndim)
+
+    # Create projector
+    vol_interp = vol_data.space.interp
+    proj_id = astra_projector(vol_interp, vol_geom, proj_geom, ndim,
+                              impl='cpu')
+
+    # Create algorithm
+    algo_id = astra_algorithm('forward', ndim, vol_id, sino_id, proj_id,
+                              impl='cpu')
+
+    # Run algorithm and delete it
+    astra.algorithm.run(algo_id)
+
+    # Flip detector pixels for fanflat
+    if isinstance(geometry, FanFlatGeometry):
+        out[:] = out.asarray()[::-1, ::-1]
+
+    # Delete ASTRA objects
+    astra_cleanup()
+
+    return out
+
+
+def astra_cpu_backward_projector_call(proj_data, geometry, reco_space,
+                                      out=None):
+    """Run an ASTRA backward projection on the given data using the CPU.
+
+    Parameters
+    ----------
+    proj_data : `DiscreteLpVector`
+        Projection data to which the backward projector is applied
+    geometry : `Geometry`
+        Geometry defining the tomographic setup
+    reco_space : `DiscreteLp`
+        Space to which the calling operator maps
+    out : `DiscreteLpVector` or `None`, optional
+        Vector in the reconstruction space to which the result is written.
+        Default: `None`
+
+    Returns
+    -------
+    out : ``reco_space`` element
+        Reconstruction data resulting from the application of the backward
+        projector
+    """
+    if not isinstance(proj_data, DiscreteLpVector):
+        raise TypeError('projection data {!r} is not a `DiscreteLpVector` '
+                        'instance.'.format(proj_data))
+    if not isinstance(geometry, Geometry):
+        raise TypeError('geometry  {!r} is not a `Geometry` instance.'
+                        ''.format(geometry))
+    if not isinstance(reco_space, DiscreteLp):
+        raise TypeError('reconstruction space {!r} is not a `DiscreteLp` '
+                        'instance.'.format(reco_space))
+
+    if reco_space.grid.ndim != geometry.ndim:
+        raise ValueError('dimensions {} of reconstruction space and {} of '
+                         'geometry do not match.'.format(reco_space.grid.ndim,
+                                                         geometry.ndim))
+
+    ndim = proj_data.space.grid.ndim
```

Comment:
I think 

```python
ndim = proj_data.space.ndim
```

is valid, possibly even

```python
ndim = proj_data.ndim
```

If they are not, they should be made valid.